logger/logger.go

```diff
@@ -0,0 +1,79 @@
+package logger
+
+import (
+	"fmt"
+	"log"
+)
+
+type Logger interface {
+	Debug(msg string)
+	Debugf(format string, args ...interface{})
+	Info(msg string)
+	Infof(format string, args ...interface{})
+	Warn(msg string)
+	Warnf(format string, args ...interface{})
+	Error(msg string)
+	Errorf(format string, args ...interface{})
+}
+
+func init() {
+	internalLogger = &LoggerImpl{}
+}
+
+var internalLogger Logger
+
+func InjectLogger(l Logger) {
```

Comment:
I would look for a way to control logger injection and access only through the external library configuration function

Reply:
I may be able to move this to an internal pkg